UIforETW/UIforETWDlg.cpp

```diff
@@ -51,13 +863,16 @@ void outputPrintf(_Printf_format_string_ const wchar_t* pFormat, ...)
 	va_end(args);
 }
 
-void CUIforETWDlg::vprintf(const wchar_t* pFormat, va_list args)
+void CUIforETWDlg::vprintf(PCWSTR pFormat, va_list args)
 {
 	wchar_t buffer[5000];
 	_vsnwprintf_s(buffer, _TRUNCATE, pFormat, args);
 
-	for (const wchar_t* pBuf = buffer; *pBuf; ++pBuf)
+	for (PCWSTR pBuf = buffer; *pBuf; ++pBuf)
 	{
+		ATLASSERT(pBuf < (buffer + ARRAYSIZE(buffer)));

```

Comment:
It's  funny that you mention allocation functions/large image buffers, because I remember a problem in chrome that I hit a while ago, where code depended on `base::UncheckedMalloc` returning a `NULL` if it failed to allocate.

On Windows, when `malloc` fails (`base::UncheckedMalloc` just calls `malloc`), the whole process terminates, which is ***EXACTLY THE OPPOSITE*** of what the documentation for `base::UncheckedMalloc` says.

Thus, `SkMallocPixelRef` would try to allocate an absurdly large buffer, and the whole browser would explode.

https://code.google.com/p/chromium/issues/detail?id=360681

/digression

Steve McConnell's books are now on my mental "to read" list. Sounds interesting.